From 784e4fcfb2c8bb1f203ff2ad0933527aa3c5e870 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 7 Jan 2026 11:08:53 +0100
Subject: [PATCH 05/18] Use a pre-built eigen dependency
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Do not link to libeigen, it's a header only dependency.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 hailort/cmake/external/eigen.cmake | 27 +--------------------------
 hailort/hailortcli/CMakeLists.txt  |  1 -
 2 files changed, 1 insertion(+), 27 deletions(-)

diff --git a/hailort/cmake/external/eigen.cmake b/hailort/cmake/external/eigen.cmake
index 1bb1f66..3912b79 100644
--- a/hailort/cmake/external/eigen.cmake
+++ b/hailort/cmake/external/eigen.cmake
@@ -1,28 +1,3 @@
 cmake_minimum_required(VERSION 3.11.0)
 
-include(FetchContent)
-
-FetchContent_Declare(
-    eigen
-    GIT_REPOSITORY https://gitlab.com/libeigen/eigen
-    GIT_TAG 3147391d946bb4b6c68edd901f2add6ac1f31f8c # Version 3.4.0
-    GIT_SHALLOW TRUE
-    SOURCE_DIR ${HAILO_EXTERNAL_DIR}/eigen-src
-    SUBBUILD_DIR ${HAILO_EXTERNAL_DIR}/eigen-subbuild
-)
-
-
-# https://stackoverflow.com/questions/65527126/disable-install-for-fetchcontent
-FetchContent_GetProperties(eigen)
-if(NOT eigen_POPULATED)
-    FetchContent_Populate(eigen)
-    option(EIGEN_BUILD_DOC OFF)
-    option(BUILD_TESTING OFF)
-    option(EIGEN_LEAVE_TEST_IN_ALL_TARGET OFF)
-    option(EIGEN_BUILD_PKGCONFIG OFF)
-    option(CMAKE_Fortran_COMPILER OFF)
-
-    if (NOT HAILO_EXTERNALS_EXCLUDE_TARGETS)
-        add_subdirectory(${eigen_SOURCE_DIR} ${eigen_BINARY_DIR} EXCLUDE_FROM_ALL)
-    endif()
-endif()
\ No newline at end of file
+find_package(Eigen3 REQUIRED NO_MODULE)
diff --git a/hailort/hailortcli/CMakeLists.txt b/hailort/hailortcli/CMakeLists.txt
index 2c1f6f6..203ca98 100644
--- a/hailort/hailortcli/CMakeLists.txt
+++ b/hailort/hailortcli/CMakeLists.txt
@@ -72,7 +72,6 @@ target_link_libraries(hailortcli
     CLI11::CLI11
     nlohmann_json
     hailort_common
-    eigen
     DotWriter
     scheduler_mon_proto
     profiler_proto)
-- 
2.55.0

